File: Samples/VB/13-ApplicationModes/Form1.vb
Imports RibbonLib
Imports RibbonLib.Controls
Imports RibbonLib.Controls.Events
Imports RibbonLib.Interop

Namespace _13_ApplicationModes
	Public Enum RibbonMarkupCommands As UInteger
		cmdTabMain = 1001
		cmdGroupCommon = 1002
		cmdGroupSimple = 1003
		cmdGroupAdvanced = 1004
		cmdButtonNew = 1005
		cmdButtonOpen = 1006
		cmdButtonSave = 1007
		cmdButtonDropA = 1008
		cmdButtonDropB = 1009
		cmdButtonDropC = 1010
		cmdButtonSwitchToAdvanced = 1011
		cmdButtonSwitchToSimple = 1012
	End Enum

	Partial Public Class Form1
		Inherits Form
		Private _tabMain As RibbonTab
		Private _groupCommon As RibbonGroup
		Private _groupSimple As RibbonGroup
		Private _groupAdvanced As RibbonGroup
		Private _buttonNew As RibbonButton
		Private _buttonOpen As RibbonButton
		Private _buttonSave As RibbonButton
		Private _buttonDropA As RibbonButton
		Private _buttonDropB As RibbonButton
		Private _buttonDropC As RibbonButton
		Private _buttonSwitchToAdvanced As RibbonButton
		Private _buttonSwitchToSimple As RibbonButton

		Public Sub New()
			InitializeComponent()

			_tabMain = New RibbonTab(_ribbon, CUInt(RibbonMarkupCommands.cmdTabMain))
			_groupCommon = New RibbonGroup(_ribbon, CUInt(RibbonMarkupCommands.cmdGroupCommon))
			_groupSimple = New RibbonGroup(_ribbon, CUInt(RibbonMarkupCommands.cmdGroupSimple))
			_groupAdvanced = New RibbonGroup(_ribbon, CUInt(RibbonMarkupCommands.cmdGroupAdvanced))
			_buttonNew = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonNew))
			_buttonOpen = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonOpen))
			_buttonSave = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonSave))
			_buttonDropA = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonDropA))
			_buttonDropB = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonDropB))
			_buttonDropC = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonDropC))
			_buttonSwitchToAdvanced = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonSwitchToAdvanced))
			_buttonSwitchToSimple = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonSwitchToSimple))

			AddHandler _buttonSwitchToAdvanced.ExecuteEvent, AddressOf _buttonSwitchToAdvanced_ExecuteEvent
			AddHandler _buttonSwitchToSimple.ExecuteEvent, AddressOf _buttonSwitchToSimple_ExecuteEvent
		End Sub

		Private Sub _buttonSwitchToAdvanced_ExecuteEvent(ByVal sender As Object, ByVal e As ExecuteEventArgs)
			_ribbon.SetModes(1)
		End Sub

		Private Sub _buttonSwitchToSimple_ExecuteEvent(ByVal sender As Object, ByVal e As ExecuteEventArgs)
			_ribbon.SetModes(0)
		End Sub
	End Class
End Namespace

File: Samples/CS/13-ApplicationModes/Form1.cs
using System;
using System.Windows.Forms;

using RibbonLib;
using RibbonLib.Controls;
using RibbonLib.Controls.Events;
using RibbonLib.Interop;

namespace _13_ApplicationModes
{
    public enum RibbonMarkupCommands : uint
    {
        cmdTabMain = 1001,
        cmdGroupCommon = 1002,
        cmdGroupSimple = 1003,
        cmdGroupAdvanced = 1004,
        cmdButtonNew = 1005,
        cmdButtonOpen = 1006,
        cmdButtonSave = 1007,
        cmdButtonDropA = 1008,
        cmdButtonDropB = 1009,
        cmdButtonDropC = 1010,
        cmdButtonSwitchToAdvanced = 1011,
        cmdButtonSwitchToSimple = 1012,
    }

    public partial class Form1 : Form
    {
        private RibbonTab _tabMain;
        private RibbonGroup _groupCommon;
        private RibbonGroup _groupSimple;
        private RibbonGroup _groupAdvanced;
        private RibbonButton _buttonNew;
        private RibbonButton _buttonOpen;
        private RibbonButton _buttonSave;
        private RibbonButton _buttonDropA;
        private RibbonButton _buttonDropB;
        private RibbonButton _buttonDropC;
        private RibbonButton _buttonSwitchToAdvanced;
        private RibbonButton _buttonSwitchToSimple;

        public Form1()
        {
            InitializeComponent();

            _tabMain = new RibbonTab(_ribbon, (uint)RibbonMarkupCommands.cmdTabMain);
            _groupCommon = new RibbonGroup(_ribbon, (uint)RibbonMarkupCommands.cmdGroupCommon);
            _groupSimple = new RibbonGroup(_ribbon, (uint)RibbonMarkupCommands.cmdGroupSimple);
            _groupAdvanced = new RibbonGroup(_ribbon, (uint)RibbonMarkupCommands.cmdGroupAdvanced);
            _buttonNew = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonNew);
            _buttonOpen = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonOpen);
            _buttonSave = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonSave);
            _buttonDropA = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonDropA);
            _buttonDropB = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonDropB);
            _buttonDropC = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonDropC);
            _buttonSwitchToAdvanced = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonSwitchToAdvanced);
            _buttonSwitchToSimple = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonSwitchToSimple);

            _buttonSwitchToAdvanced.ExecuteEvent += new EventHandler<ExecuteEventArgs>(_buttonSwitchToAdvanced_ExecuteEvent);
            _buttonSwitchToSimple.ExecuteEvent += new EventHandler<ExecuteEventArgs>(_buttonSwitchToSimple_ExecuteEvent);
        }
        
        void _buttonSwitchToAdvanced_ExecuteEvent(object sender, ExecuteEventArgs e)
        {
            _ribbon.SetModes(1);
        }

        void _buttonSwitchToSimple_ExecuteEvent(object sender, ExecuteEventArgs e)
        {
            _ribbon.SetModes(0);
        }
    }
}
